compare-result.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import sys
+
+def create_parser():
+    """Command line parser."""
+
+    parser = argparse.ArgumentParser(
+        description='Compare CBMC property checking results.'
+    )
+
+    parser.add_argument(
+        'result1',
+        help="""
+        JSON result output of make-result or cbmc-viewer.
+        """
+    )
+    parser.add_argument(
+        'result2',
+        help="""
+        JSON result output of make-result or cbmc-viewer.
+        """
+    )
+
+    return parser
+
+def compare(result1, result2, full=False):
+    try:
+        dict1 = result1['viewer-result']
+        dict2 = result2['viewer-result']
+
+        keys = set([*dict1.keys(), *dict2.keys()])
+        if not full:
+            keys.remove('program')
+            keys.remove('status')
+
+        return all([dict1[key] == dict2[key] for key in keys])
+    except KeyError:
```

Comment:
Maybe this is pythonic, in which case keep doing it. But using exceptions for ordinary control flow seems weird to my non python native brain.